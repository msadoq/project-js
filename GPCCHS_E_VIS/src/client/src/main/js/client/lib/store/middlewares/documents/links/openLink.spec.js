import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import createOpenLinkMiddleware from './openLink';

describe('middlewares/createOpenLinkMiddleware', () => {
  const documentManager = {
    openPage: payload => ({ type: 'OPEN_PAGE', payload }),
    openView: (payload, pageId) => ({ type: 'OPEN_VIEW', payload: { ...payload, pageId } }),
    readDocumentType: ({ path }, cb) => {
      if (path === '/page' || path === '/p1') {
        return cb(null, 'Page');
      }
      if (path === '/textview' || path === '/v2') {
        return cb(null, 'TextView');
      }
      if (path === '/error') {
        return cb(new Error('Error readDocumentType'));
      }
      return cb(null, 'unknownType');
    },
  };

  const mockStore = configureMockStore([thunk, createOpenLinkMiddleware(documentManager)]);
  const store = mockStore({
    timebars: { tb1: {} },
    windows: { w1: { pages: ['p1'] } },
    hsc: { focusWindow: 'w1' },
    views: {
      v1: {
        uuid: 'v1',
        absolutePath: '/path/to/view.json',
        links: [
          { path: '/page' },
          { path: '/textview' },
          { path: '/error' },
          { path: '/v2' },
          { path: '/p1' },
          { path: '/unknown' },
        ],
      },
      v2: {
        uuid: 'v2',
        absolutePath: '/v2',
      },
    },
    pages: {
      p1: {
        views: ['v2'],
        uuid: 'p1',
        absolutePath: '/p1',
      },
    },
  });
  afterEach(() => {
    store.clearActions();
  });

  const openLink = linkId => ({
    type: 'WS_ASK_OPEN_LINK',
    payload: {
      viewId: 'v1',
      linkId,
    },
  });

  test('open a page', () => {
    store.dispatch(openLink(0));
    expect(store.getActions()).toMatchSnapshot();
  });
  test('open a view', () => {
    store.dispatch(openLink(1));
    expect(store.getActions()).toMatchSnapshot();
  });
  test('Error : Message on readDocumentType error', () => {
    store.dispatch(openLink(2));
    expect(store.getActions()).toMatchSnapshot();
  });
  test('Error : Message on unknown link', () => {
    store.dispatch(openLink(999));
    expect(store.getActions()).toMatchSnapshot();
  });
  test('Error: Message on type error', () => {
    store.dispatch(openLink(5));
    expect(store.getActions()).toMatchSnapshot();
  });
  test('open a linked view already opened (focus)', () => {
    store.dispatch(openLink(3));
    expect(store.getActions()).toMatchSnapshot();
  });
  test('open a linked page already opened (focus)', () => {
    store.dispatch(openLink(4));
    expect(store.getActions()).toMatchSnapshot();
  });
  test('Other action', () => {
    const otherActions = { type: 'WS_OTHER_ACTION' };
    store.dispatch(otherActions);
    expect(store.getActions()).toEqual([{ type: 'WS_OTHER_ACTION' }]);
  });
});
