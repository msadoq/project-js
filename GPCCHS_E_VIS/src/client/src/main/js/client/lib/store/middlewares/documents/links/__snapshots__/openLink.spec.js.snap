// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`middlewares/makeOpenLinkMiddleware Error : Message on readDocumentType error 1`] = `
Array [
  Object {
    "payload": Object {
      "linkId": 2,
      "viewId": "v1",
    },
    "type": "WS_ASK_OPEN_LINK",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Error readDocumentType",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`middlewares/makeOpenLinkMiddleware Error : Message on unknown link 1`] = `
Array [
  Object {
    "payload": Object {
      "linkId": 999,
      "viewId": "v1",
    },
    "type": "WS_ASK_OPEN_LINK",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "v1",
      "messages": Array [
        Object {
          "content": "Unknown link",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`middlewares/makeOpenLinkMiddleware Error: Message on type error 1`] = `
Array [
  Object {
    "payload": Object {
      "linkId": 5,
      "viewId": "v1",
    },
    "type": "WS_ASK_OPEN_LINK",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "containerId": "global",
      "messages": Array [
        Object {
          "content": "Error, unknown type 'unknownType'",
        },
      ],
      "type": "danger",
    },
    "type": "WS_MESSAGE_ADD",
  },
]
`;

exports[`middlewares/makeOpenLinkMiddleware open a linked page already opened (focus) 1`] = `
Array [
  Object {
    "payload": Object {
      "linkId": 4,
      "viewId": "v1",
    },
    "type": "WS_ASK_OPEN_LINK",
  },
  Object {
    "payload": Object {
      "pageId": "p1",
      "windowId": "w1",
    },
    "type": "WS_WINDOW_PAGE_FOCUS",
  },
]
`;

exports[`middlewares/makeOpenLinkMiddleware open a linked view already opened (focus) 1`] = `
Array [
  Object {
    "payload": Object {
      "linkId": 3,
      "viewId": "v1",
    },
    "type": "WS_ASK_OPEN_LINK",
  },
  Object {
    "payload": Object {
      "pageId": "p1",
      "windowId": "w1",
    },
    "type": "WS_WINDOW_PAGE_FOCUS",
  },
]
`;

exports[`middlewares/makeOpenLinkMiddleware open a page 1`] = `
Array [
  Object {
    "payload": Object {
      "linkId": 0,
      "viewId": "v1",
    },
    "type": "WS_ASK_OPEN_LINK",
  },
  Object {
    "payload": Object {
      "path": "/page",
      "viewFolder": "/path/to",
      "windowId": "w1",
    },
    "type": "OPEN_PAGE",
  },
]
`;

exports[`middlewares/makeOpenLinkMiddleware open a view 1`] = `
Array [
  Object {
    "payload": Object {
      "linkId": 1,
      "viewId": "v1",
    },
    "type": "WS_ASK_OPEN_LINK",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "page": Object {
        "timebarUuid": "tb1",
      },
      "windowId": "w1",
    },
    "type": "WS_PAGE_ADD_BLANK",
  },
  Object {
    "meta": Object {
      "jestSerialized": true,
    },
    "payload": Object {
      "path": "/textview",
      "viewFolder": "/path/to",
    },
    "type": "OPEN_VIEW",
  },
]
`;
